packages/camera/camera/android/src/main/java/io/flutter/plugins/camera/Camera.java

```diff
@@ -79,6 +79,24 @@
   void onError(String errorCode, String errorMessage);
 }
 
+/** A mockable wrapper for CameraDevice calls. */
+interface CameraDeviceWrapper {
```

Comment:
I think this is a Delegate. `CameraDeviceDelegate` works.

https://en.wikipedia.org/wiki/Delegation_pattern

Reply:
I don't see how this is a delegate; nothing is being delegated to it. It's just an interface to allow mocking a `CameraDevice`

Reply:
oh I see, DefaultCameraDeviceWrapper delegates to CameraDevice